sktime/performance_metrics/forecasting/probabilistic/_classes.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+import numpy as np
+
+# TODO: add formal tests
+import pandas as pd
+from sklearn.metrics._regression import _check_reg_targets
+
+from sktime.performance_metrics.forecasting._classes import _BaseForecastingErrorMetric
+
+
+def _check_y_pred_type(metric_object, y_pred):
+    input_type = metric_object.get_tag("scitype:y_pred")
+    if input_type == "quantiles":
+        if isinstance(y_pred, pd.DataFrame):
+            return y_pred
+        # if distribution object then get the relevant quantiles and return.
+
+
+class _BaseProbForecastingErrorMetric(_BaseForecastingErrorMetric):
+    """Base class for probabilistic forecasting error metrics in sktime.
+
+    Extends sktime's BaseMetric to the forecasting interface. Forecasting error
+    metrics measure the error (loss) between forecasts and true values. Lower
+    values are better.
+    """
+
+    _tags = {"scitype:y_pred": "quantiles"}
+
+    def __init__(self, func, name=None, multioutput="uniform_average"):
```

Comment:
what is the "func" being used for, is that really necessary?

Reply:
The `func` contains the underlying logic implementation for the metric. It is necessary as I am inheriting from the previous metric class which included it. It can be useful if you need two different metric classes that use the same function in different ways.

Reply:
hm, ok - is there a way to make it optional?

I'm thinking about extenders, who will probably fund it very confusing tha `pinball_loss` is passed in two places, and `func` isn't really used in the child, `PinballLoss`.
For instance, why can `_evaluate` not take the place of `func`? Or the other way round?